src/clj/game/cards/icebreakers.clj

```diff
@@ -251,40 +251,48 @@
+                                                          " to break "
+                                                          (quantify n (str ice-type " subroutine")))))))}
                {:label "Match strength of currently encountered ice"
                 :req (req (and current-ice
                                (> (ice-strength state side current-ice)
-                                  (or (:current-strength card) (:strength card)))))
+                                  (or (:current-strength card)
+                                      (:strength card)))))
                 :effect (req (wait-for (resolve-ability
                                          state side
                                          (pick-virus-counters-to-spend
                                            (- (ice-strength state side current-ice)
-                                              (or (:current-strength card) (:strength card))))
+                                              (or (:current-strength card)
+                                                  (:strength card))))
                                          card nil)
-                                       (if-let [msg (:msg async-result)]
-                                         (do (system-msg state :runner (str "spends " msg " to add "
-                                                                            (:number async-result) " strength"))
-                                             (dotimes [_ (:number async-result)]
-                                               (pump state side (get-card state card) 1))))))}
+                                       (when-let* [msg (:msg async-result)
+                                                   n (:number async-result)]
+                                         (dotimes [_ n]
+                                           (pump state side (get-card state card) 1))
+                                         (system-msg state :runner
+                                                     (str "spends " msg
+                                                          " to add " n
+                                                          " strength")))))}
                {:label "Add strength"
-                :effect (req (wait-for
-                               (resolve-ability
-                                 state side (pick-virus-counters-to-spend) card nil)
-                               (if-let [msg (:msg async-result)]
-                                 (do (system-msg state :runner (str "spends " msg" to add "
-                                                                    (:number async-result)
-                                                                    " strength"))
-                                     (dotimes [_ (:number async-result)]
-                                       (pump state side (get-card state card) 1))))))}]})
+                :effect (req (wait-for (resolve-ability
```

Comment:
This code is very similar to the code above. I wonder if it can be combined in a nice way..